File: frontend/src/store/modules/remote-servers.js
import * as actionsList from '@/store/actions-list'
import httpClient from '@/services/http-client'
import RemoteServer from '@/models/RemoteServer'

const types = {
    REFRESH_REMOTE_SERVERS: 'REFRESH_REMOTE_SERVERS'
}

const state = {
    remoteServers: []
}

const getters = {
    remoteServers: state => state.remoteServers
}

const mutations = {
    [types.REFRESH_REMOTE_SERVERS] (state, remoteServers) {
        state.remoteServers = remoteServers
    }
}

const actions = {
    [actionsList.GET_REMOTE_SERVERS] ({ state, commit, dispatch }) {
        return new Promise((resolve, reject) => {
            httpClient.get('/api/remote-servers')
                .then((remoteServers) => {
                    var newRemoteServers = []

                    for (var i = 0, nb = remoteServers.length; i < nb; i++) {
                        newRemoteServers.push(new RemoteServer(
                            remoteServers[i].id,
                            remoteServers[i].name,
                            remoteServers[i].host,
                            remoteServers[i].username,
                            remoteServers[i].sshKeyPath
                        ))
                    }

                    commit(types.REFRESH_REMOTE_SERVERS, remoteServers)
                    resolve()
                }).catch((msg) => {
                    reject(new Error(msg))
                })
        })
    }
}

export default {
    state,
    actions,
    mutations,
    getters
}

File: backend/storage/remote_servers.go
package storage

type RemoteServer struct {
	Id         int    `json:"id"`
	Host       string `json:"host"`
	Username   string `json:"username"`
	SshKeyPath string `json:"ssh_key_path"`
}

// GetRemoteServers returns the list of available remote servers
func GetRemoteServers() ([]*RemoteServer, error) {
	s, err := getStorage()

	if err != nil {
		return nil, err
	}

	if s.RemoteServers == nil {
		return make([]*RemoteServer, 0), nil
	}

	return s.RemoteServers, nil
}
